core/modules/url/pages/UrlClientDispatcher.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+  <component :is="page" />
+</template>
+
+<script>
+import { UrlDispatchMapper, _handleDispatcherNotFound } from '@vue-storefront/core/modules/url'
+import userRoutes from 'theme/router'
+import store from '@vue-storefront/store'
+import { AsyncDataLoader } from '@vue-storefront/core/lib/async-data-loader'
+
+let _matchingComponentInstance = null
+const dispatcherRoutine = (to, from, next) => {
+  UrlDispatchMapper(to).then(routeData => {
+    if (routeData) {
+      if (store.state.config.seo.useUrlDispatcher && routeData) {
+        const userRoute = userRoutes.find(r => r.name === routeData['name'])
+        if (userRoute) {
+          userRoute.component().then(rootComponent => {
+            _matchingComponentInstance = rootComponent.default
+
+            if (from.name) { // don't execute the action on the Client's side of the SSR actions
+              if (_matchingComponentInstance.asyncData) {
+                AsyncDataLoader.push({
+                  execute: _matchingComponentInstance.asyncData
+                })
+              }
+              if (_matchingComponentInstance.preAsyncData) {
+                AsyncDataLoader.push({
+                  execute: _matchingComponentInstance.preAsyncData
+                })
+              }
+              if (_matchingComponentInstance.mixins) {
+                _matchingComponentInstance.mixins.map(m => {
+                  if (m.preAsyncData) {
+                    AsyncDataLoader.push({ // this is an example showing how to call data loader from another module
+                      execute: m.preAsyncData
+                    })
+                  }
+                  if (m.asyncData) {
+                    AsyncDataLoader.push({ // this is an example showing how to call data loader from another module
+                      execute: m.asyncData
+                    })
+                  }
+                })
+              }
+            }
+            next()
+          })
+        } else {
+          _handleDispatcherNotFound(routeData['name'])
+        }
+      } else {
+        _handleDispatcherNotFound(null)
+      }
+    } else {
+      next()
+    }
+  })
+}
+
+export default {
+  computed: {
+    page: () => {
+      return _matchingComponentInstance
```

Comment:
this shouldn't be a computer property + you shouldn't use arrow functions in computed properties